### PR TITLE
[feature](hive) support hive write text table

### PR DESCRIPTION
**Pull request by suxiaogang223**
## Proposed changes

1. Support write hive text table
2. Add SessionVariable `hive_text_compression` to write compressed hive text table
3. Supported compression type: gzip, bzip2, snappy, lz4, zstd

### CODE REVIEW

**Comment by suxiaogang223** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Comment by suxiaogang223** [comment]
run buildall

File: be/src/util/slice.h [hunk 1]
```diff
@@ -358,6 +358,10 @@ class OwnedSlice : private Allocator<false, false, false, DefaultMemoryAllocator
         return *this;
     }
 
+    // disable copy constructor and copy assignment
```
[the remark this answers is not in the document]
Comment (suxiaogang223) [reply be/src/util/slice.h:361]: just follow the clang-tidy worrying
https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-equals-delete.html

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 2]
```diff
@@ -1111,6 +1113,9 @@ public class SessionVariable implements Serializable, Writable {
                     "set the number of sort phases 1 or 2. if set other value, let cbo decide the sort type"})
     public int sortPhaseNum = 0;
 
+    @VariableMgr.VarAttr(name = HIVE_TEXT_COMPRESSION, needForward = true)
+    private String hiveTextCompression = "uncompressed";
```
[the remark this answers is not in the document]
Comment (suxiaogang223) [reply fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:1117]: It is used to determine the compressed file format written to the hive_text table.

**Comment by suxiaogang223** [comment]
run buildall

**Comment by suxiaogang223** [comment]
run buildall

**Comment by suxiaogang223** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Comment by suxiaogang223** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by morningman — APPROVED** [review]

**Review by wuwenchi — APPROVED** [review]

**Pull request merged**